src/supautils.c

```diff
@@ -367,15 +367,9 @@ supautils_hook(PROCESS_UTILITY_PARAMS)
 					comfirm_reserved_roles(created_role);
 
 					// Allow bypassrls attribute if using `privileged_role`.
-					if (privileged_role == NULL) {
+					if (privileged_role != NULL &&
+						GetUserId() == get_role_oid(privileged_role, true))
 						bypassrls_is_allowed = false;
-					}
-					if (!OidIsValid(get_role_oid(privileged_role, true))) {
-						bypassrls_is_allowed = false;
-					}
-					if (GetUserId() != get_role_oid(privileged_role, false)) {
-						bypassrls_is_allowed = false;
-					}
```

Comment:
Oops, just merged https://github.com/supabase/supautils/pull/40 which contained a refactor of these lines.

Please make sure to rebase.

Reply:
Thanks for the review!
I rebased the code. Could you review the rebased one?